README.md

```diff
@@ -43,32 +43,34 @@ for definition of maturity levels).
 
 Components marked `Stable` provide the following guarantees:
 
-- Field types will not change.
-- Field numbers will not change.
-- Numbers assigned to enum choices will not change.
-- Service names and service package names will not change.
-- Service operation names, parameter and return types will not change.
+- Field types, numbers and names will not change.
```

Comment:
Why not allowing to remove a field? Since we allow sending "unknown" fields, removing a field is not a wire breaking change.

Not that I want to do this, but I want to clarify the "boundaries" and this is only related to generated code.

Reply:
From the wire perspective you are right, in theory we could do this and have an explanation that for receivers it is equivalent to the previous state when sender didn't include the field. In practice it would be very difficult to do without breaking some functionality that depends on the field. We don't really have unnecessary fields :-)
  
However because we also aim to maintain symbolic compatibility remove a field would breaking any existing code that uses it. So from this perspective it definitely should be prohibited.

Reply:
Removing a field without reserving the ID brings the possibility the ID would be used later with a non-compatible data. If we reserve the field, there is no real difference between keeping it (marked deprecated) and removing it.

Reply:
>If we reserve the field, there is no real difference between keeping it (marked deprecated) and removing it.

There is a difference for the generated code. It is a breaking change to remove a field.